src/frontend/web_application/src/scenes/ContactBook/presenter.jsx

```diff
@@ -11,6 +11,15 @@ import './style.scss';
 const defaultSortDir = 'ASC';
 const defaultSortView = 'title';
 
+function generateStateFromProps(props) {
+  const { contacts } = props;
+  const tags = [];
+  contacts.map(contact => contact.tags.map(tag => tags.push(tag)));
```

Comment:
If you wanted to make it less imperative, and embrace a little more the `.map` API style:

```js
const tags = [].concat(...contacts.map(contact => contact.tags));
```
If you wish to also make it an array of unique values:

```js
const tags =[...new Set([].concat(...contacts.map(contact => contact.tags)))];
```